src/compiler/types.ts

```diff
@@ -4959,13 +4963,13 @@ namespace ts {
          * If resolveModuleNames is implemented then implementation for members from ModuleResolutionHost can be just
          * 'throw new Error("NotImplemented")'
          */
-        resolveModuleNames?(moduleNames: string[], containingFile: string, reusedNames?: string[]): (ResolvedModule | undefined)[];
+        resolveModuleNames?(moduleNames: string[], containingFile: string, reusedNames?: string[], redirectedReference?: ResolvedProjectReference): (ResolvedModule | undefined)[];

```

Comment:
when resolving module names in the root project, there will be no `ResolvedProjectReference`. Therefore a custom CompilerHost has no access to the CompilerOptions of that project.

Reply:
That is not changed since it was original API as well. In general host already knows root compiler options. Eg. https://github.com/TypeStrong/ts-loader/blob/master/src/servicesHost.ts#L498

Reply:
> In general host already knows root compiler options.

Correct, although that only works when building exactly one project with that host. The current implementation of `tsbuild.ts` (I know it's still considered internal) uses the same `SolutionBuilderHost` to build all projects. That host cannot know about the compilerOptions of all projects. And since tsbuild can even handle multiple tsconfig.json files at once, there's no way the host could ever know the correct compilerOptions.

Reply:
I don't think we want to change that. (the default compilerHost is for specific compilerOptions) Infact we want to do that from ts build as well but that's a todo for little later.